src/ui-test/camelk_java_debug_test.ts

```diff
@@ -0,0 +1,201 @@
+import { assert } from 'chai';
+import path = require('path');
+import {
+	ActivityBar,
+	CustomTreeSection,
+	DebugToolbar,
+	EditorView,
+	InputBox,
+	SideBarView,
+	TextEditor,
+	ViewItem,
+	VSBrowser,
+	Workbench
+} from 'vscode-extension-tester';
+import { prepareEmptyTestFolder } from './utils/resourcesUtils';
+
+const TEST_FOLDER = '../../../testFolder';
+const WORKSPACE_FOLDER = path.join(__dirname, TEST_FOLDER);
+
+const START_DEBUG_LABEL = 'Start Java debugger on Camel K integration';
+const REMOVE_INTEGRATION_LABEL = 'Remove Apache Camel K Integration';
+
+
+describe('Tooling for Apache Camel K extension', function () {
+
+	before(async function () {
+		this.timeout(200000);
+		await prepareTempWorkspaceForTests(WORKSPACE_FOLDER);
+		await VSBrowser.instance.driver.sleep(3000);
+	});
+	
+	describe('Java Debug', function () {
+		
+		const INTEGRATION_LABEL = 'java-debug-test';
+		const INTEGRATION_FILE = 'JavaDebugTest';
+
+		before(async function (){
+			this.timeout(20000);
+			await createIntegration(INTEGRATION_FILE);
+			await startIntegrationOnCurrentFile();
+			await VSBrowser.instance.driver.sleep(5000);
+		})
+
+		it('Check Java Debug available', async function () {
+			this.timeout(20000);
+
+			const item = await findIntegrationOnSideBar(INTEGRATION_LABEL);
+			const menu = await item.openContextMenu();
+
+			assert.isTrue(await menu.hasItem(START_DEBUG_LABEL));
+		});
+
+		it.skip('Check Java Debug stops at breakpoint', async function() {
+			//Currently failing because "No Java Debugger configured" error window
```

Comment:
I tried just that after Dominik told me but it didn't appear to be working (the install from marketplace method). If I got into the instance I could see the extension wasn't installed.

In any case, even installing the extension didn't solve the configuration problem (tried installing it manually in the vscode test instance while it was slept, and then leaving the window at where the tests expected after making sure the extension was installed, but there was still the same error popping of "no Java Debug configured")

Reply:
I do not understand. Do yuou mean that manually you were not able to have the feature working at all?

Reply:
Could you please try to different setup for uitests_runner? It looks like `setupAndRun` will override installed all vscode instance setup which was done till that time.

e.g. something similar this one:
```js
const tester = new ExTester(storageFolder, releaseType, extensionFolder);
await tester.downloadCode(process.env.CODE_VERSION);
await tester.downloadChromeDriver(process.env.CODE_VERSION);
await tester.installFromMarketplace('redhat.java');
await tester.installFromMarketplace('vscjava.vscode-java-debug');
await tester.runTests('out/src/ui-test/*extension_test.js', { vscodeVersion: process.env.CODE_VERSION });
```

Reply:
I'll try this on Monday, at least manually to see the extension is actually there in the marketplace, but I dont think Ill implement this in the tests themselves until we are unblocked.

Reply:
there was missing the Camel K extension with code in previous comment, I think it will be more somethign like:

```
	const tester = new ExTester(storageFolder, releaseType, extensionFolder);
	await tester.downloadCode(process.env.CODE_VERSION);
	await tester.downloadChromeDriver(process.env.CODE_VERSION);
	await tester.installFromMarketplace('redhat.java');
	await tester.installFromMarketplace('vscjava.vscode-java-debug');
	await tester.installVsix();
	await tester.runTests('out/src/ui-test/*java_debug_test.js', { vscodeVersion: process.env.CODE_VERSION });
```
	
notice the `await tester.installVsix();`